### PR TITLE
Fix for case where `pc_features.npy` does not exist

### PR DESCRIPTION
**Pull request by kabilar**
- When Kilosort does not output a `pc_features.npy` file, set `CuratedClustering.Unit::spike_depths` attribute to null.
- Will create new tag, release, and PyPI publication once this is merged.

### CODE REVIEW

File: element_array_ephys/readers/kilosort.py [hunk 1]
```diff
@@ -127,19 +129,20 @@ def get_best_channel(self, unit):
...
+
+            # ---- compute center of mass of these features (spike depths) ----
+
+            # which channels for each spike?
+            spk_feature_ind = self.data['pc_feature_ind'][self.data['spike_templates'], :]
+            # ycoords of those channels?
+            spk_feature_ycoord = ycoords[spk_feature_ind]
+            # center of mass is sum(coords.*features)/sum(features)
+            self._data['spike_depths'] = (np.sum(spk_feature_ycoord * pc_features**2, axis=1)
+                                        / np.sum(pc_features**2, axis=1))
```
Comment (ttngu207) [inline element_array_ephys/readers/kilosort.py:146]:
```suggestion
else:
    self._data['spike_depths'] = None
```

File: element_array_ephys/ephys_precluster.py [hunk 2]
```diff
@@ -667,7 +667,8 @@ def make(self, key):
         # -- Spike-sites and Spike-depths --
         spike_sites = np.array([channel2electrodes[s]['electrode']
                                 for s in kilosort_dataset.data['spike_sites']])
-        spike_depths = kilosort_dataset.data['spike_depths']
+        spike_depths = kilosort_dataset.data['spike_depths'] \
+                        if 'pc_features' in kilosort_dataset.data else None
```
Comment (ttngu207) [inline element_array_ephys/ephys_precluster.py:671]:
```suggestion
        spike_depths = kilosort_dataset.data['spike_depths']
```

File: element_array_ephys/ephys_acute.py [hunk 3]
```diff
@@ -533,7 +533,8 @@ def make(self, key):
         # -- Spike-sites and Spike-depths --
         spike_sites = np.array([channel2electrodes[s]['electrode']
                                 for s in kilosort_dataset.data['spike_sites']])
-        spike_depths = kilosort_dataset.data['spike_depths']
+        spike_depths = kilosort_dataset.data['spike_depths'] \
+                        if 'pc_features' in kilosort_dataset.data else None
```
Comment (kabilar) [inline element_array_ephys/ephys_acute.py:537]:
```suggestion
        spike_depths = kilosort_dataset.data['spike_depths']
```

File: element_array_ephys/ephys_chronic.py [hunk 4]
```diff
@@ -532,7 +532,8 @@ def make(self, key):
         # -- Spike-sites and Spike-depths --
         spike_sites = np.array([channel2electrodes[s]['electrode']
                                 for s in kilosort_dataset.data['spike_sites']])
-        spike_depths = kilosort_dataset.data['spike_depths']
+        spike_depths = kilosort_dataset.data['spike_depths'] \
+                        if 'pc_features' in kilosort_dataset.data else None
```
Comment (kabilar) [inline element_array_ephys/ephys_chronic.py:536]:
```suggestion
        spike_depths = kilosort_dataset.data['spike_depths']
```

**Comment by kabilar** [comment]
Thanks @ttngu207!  I have implemented your suggestions.

**Pull request merged**